apps/etterlatte-saksbehandling-ui/server/src/middleware/auth.ts

```diff
@@ -39,6 +39,22 @@ export const authenticateUser = (req: Request, res: Response, next: NextFunction
     logger.error('Feil ved validering av token', e)
     return res.status(401).send('ugyldig token')
   }
+  const NAVident = parsedToken.NAVident
+  logger.info(`Navident logger på ${NAVident} cluster-name ${process.env.NAIS_CLUSTER_NAME}`)
+  if (process.env.NAIS_CLUSTER_NAME === 'prod-gcp') {
+    const saksbehandlere = process.env.saksbehandlere?.split(':')
+    if (!saksbehandlere?.includes(NAVident)) {
+      logger.error(`Saksbehandler utenfor scope forsøke å logge på, ident: ${NAVident}`)
+      return res.status(401).send('Saksbehandler mangler tilgang')
+    }
+  } else if (process.env.NAIS_CLUSTER_NAME === 'dev-gcp') {
+    logger.info(`saksbehandlere fra secret: ${process.env.saksbehandlere}`)
+    const saksbehandlere = process.env.saksbehandlere?.split(':')
+    if (!saksbehandlere?.includes(NAVident)) {
+      logger.error(`Saksbehandler utenfor scope forsøke å logge på, ident: ${NAVident}`)
+      return res.status(401).send('Saksbehandler mangler tilgang')
+    }
+  }
```

Comment:
Er det mulig å slå disse sammen?
``` 
const cluster = process.env.NAIS_CLUSTER_NAME   
  
if (['prod-gcp','dev-gcp'].includes(cluster){
  const saksbehandlere = process.env.saksbehandlere?.split(':')
  if (!saksbehandlere?.includes(NAVident)) {
    logger.error(`Saksbehandler utenfor scope forsøke å logge på, ident: ${NAVident}`)
    return res.status(401).send('Saksbehandler mangler tilgang')
  }
  if (cluster === 'dev-gcp') logger.info(`saksbehandlere fra secret: ${process.env.saksbehandlere}`)
}
```

Reply:
Absolutt! Spørsmålet er vel bare om vi skal ha dev i dev-gcp, det vet jeg ikke helt.

Reply:
Hva med å bruke en "feature toggle" så man enkelt kan skru det av og på hvis vi finner ut at vi vil ha/ikke ha det i dev?

Reply:
Vil du ta i bruk unleash?

Reply:
Kan bruke en env i nais.yaml som en start? Mulig det er litt overkill og ta i bruk unleash nå?

Reply:
[[EY-1441 Featuretoggle for tilgangsstyring](https://github.com/navikt/pensjon-etterlatte-saksbehandling/pull/398/commits/94c555b3366b625470b9ea860f2819cc27e14eb4)](https://github.com/navikt/pensjon-etterlatte-saksbehandling/pull/398/commits/94c555b3366b625470b9ea860f2819cc27e14eb4)